### ISSUE TITLE
6/15 Rogue/Data Meeting Recap

### ISSUE THREAD

**Issue by ngjo** [DoSomething/quasar#363]
**6/15 Rogue/Data Meeting Recap**

Some areas of discrepancy and next steps:
1. SMS Duplicate issue on Gambit for folks who messaged in w/ +160 characters creating multiple signups and multiple reportbacks came up again (example: https://www.dosomething.org/us/user/5007132).  This is a problem because we're over reporting  signups and reportbacks. (Issue first flagged here: https://trello.com/c/8WAc2icF/1657-duplicate-rbs-w-sms-as-the-source) 
2. We came across a user (Northstar ID: 5581ca74469c6475138b5518, UID: 2806506) who has reportbacks for a campaign, but doesn't have a signup for that campaign. This is a problem because Rogue is looking for signups. (A larger question: do we want to count reportbacks if the signup is deleted?)
3. Quasar should be ingesting `signup_id` instead of `signup_event_id` as that is the signup for a member for a given campaign. This will allow Looker to look at the `signup_id` or signup count.
4. It seems that SMS Betas are being assigned a NULL value for their UIDs - they are not being counted as a signup (I think?)
5. In Looker, for pre-Gambit campaigns, we will show the current dashboards and won't try to hook it up with Rogue data. For post-Gambit campaigns, we will show dashboards w/ Rogue data as Gambit receives signups/reportbacks which are sent to Phoenix that are in turn sent to Rogue. We made this decision because the main discrepancy for signups on the pre-Gambit campaigns are due to the de-duped Mobile Commons data. Working with this query is difficult, and we felt, unnecessary if pre-Gambit campaigns hold historic data. Here's an example of pre-Gambit, Power to the Period data: 

![fullsizerender 1](https://user-images.githubusercontent.com/2658867/27193814-4c7805e6-51ce-11e7-9b2e-c8323abfc202.jpg)

**Comment by sbsmith86** [comment]
There was one campaign where there was a discrepancy between Rogue Looker data and Rogue Prod data
![img_7859](https://user-images.githubusercontent.com/1700409/27195178-b062788e-51d3-11e7-8bfe-9f00d9fbf061.JPG)

**Comment by sbsmith86** [comment]
Documenting the Rogue queries that we ran against prod:

```
/* total signups: 3,407,695 */
SELECT COUNT(*) FROM `signups`;

/* [7249] captain obvious signups: 5107        DISTINCT: 5,107 */
/* [7549] rinse recycle signups: 38,807        DISTINCT: 38,786 */
/* [7783] mirror messages signups: 132,425     DISTINCT: 132,405 */
SELECT COUNT(DISTINCT(northstar_id))
FROM `signups` WHERE `campaign_run_id` = '7549';

/* [7249] captain obvious reportbacks: 912 */
/* [7549] rinse recycle reportbacks: 931 */
/* [7783] mirror messages reportbacks: 4,383 */
SELECT COUNT(DISTINCT(p.northstar_id)) FROM `posts` p
JOIN `signups` s ON p.signup_id = s.id
WHERE s.campaign_run_id = '7249';

/* [7249] captain obvious approved reportbacks: 827 */
/* [7549] rinse recycle approved reportbacks: 827 */
/* [7783] mirror messages approved reportbacks: 4,051 */
SELECT COUNT(DISTINCT(p.northstar_id)) FROM `posts` p
JOIN `signups` s ON p.signup_id = s.id
WHERE s.campaign_run_id = '7249'
AND p.status = 'accepted';
```

**Comment by sbsmith86** [comment]
Documenting the Phoenix queries we ran againist prod to find total reportbacks:

```
select count(distinct rb.uid)
from dosomething_reportback_file rbf
join dosomething_reportback rb on rb.rbid = rbf.rbid
join dosomething_signup s on rb.uid = s.uid AND rb.nid = s.nid AND rb.run_nid = s.run_nid
where rb.run_nid = 7549;
```

**Comment by sheyd** [comment]
@jamjensen @sbsmith86 For point 3 above, we are ingesting the `signup_id`. The mapping of `signup_id` to `signup_event_id` is just schema naming diffs. The relevant code mapping is here: https://github.com/DoSomething/quasar/blob/master/etl-scripts/QuasarCampaignActivityParsing.py#L120-L140

**Comment by sheyd** [comment]
@jamjensen Just following up on this line of questioning that we discussed IRL today, does it make sense to update this column name in the schema to `signup_id` from `signup_event_id` and update the code to make things less confusing?

**Comment by jamjensen** [comment]
Yes! As long as they're referring to the same number I think it make things
much clearer if we called it signup_id.

**Comment by jamjensen** [comment]
Here's an overview of campaign activity data (both for pre and post gambit campaigns) split by Phoenix Looker, Rogue Looker and Rogue Prod. I added columns for (1) total number of web signups; (2) total number of reportbacks submitted via sms; and (3) total number of approved reportbacks submitted via sms so that we can better understand the source of data discrepancies. 

https://docs.google.com/a/dosomething.org/spreadsheets/d/1uDK79XkA589s1y06lGFo03wX3KuJxCKmPJEzF4VLRKA/edit?usp=sharing 

**Additional thoughts**
*Pre-Gambit*
- Difference in number of signups and reportbacks can largely (but not entirely)be explained by the fact that Rogue is not pulling in Mobile Commons data. 

*Post-Gambit*
- delta in total signups is <10 for each of the four campaigns I looked at
- delta in total reportbacks ranges from 250 to 1.
